src/db.c

```diff
@@ -1150,30 +1154,39 @@ void db_col_release(struct db_filter_col *col)
 }
 
 /**
- * Validate the seccomp action
- * @param action the seccomp action
+ * Validate a filter collection
+ * @param col the seccomp filter collection
+ *
+ * This function validates a seccomp filter collection.  Returns zero if the
+ * collection is valid, negative values on failure.
  *
- * Verify that the given action is a valid seccomp action; return zero if
- * valid, -EINVAL if invalid.
  */
-int db_action_valid(uint32_t action)
+int db_col_valid(struct db_filter_col *col)
 {
-	if (sys_chk_seccomp_action(action) == 1)
+	if (col != NULL && col->state == _DB_STA_VALID && col->filter_cnt > 0)
 		return 0;
 	return -EINVAL;
 }
 
 /**
- * Validate a filter collection
- * @param col the seccomp filter collection
- *
- * This function validates a seccomp filter collection.  Returns zero if the
- * collection is valid, negative values on failure.
+ * Validate the seccomp action
+ * @param action the seccomp action
```

Comment:
nitpick - we should add @param for col as well

Reply:
Another good catch, thanks.

Since this is the only remaining issue and it's a function header comment for an internal function, I'm simply going to go ahead and merge this after I fix it up, another review round seems a little silly at this point.  Thanks for all the review help.